internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -294,18 +353,19 @@ func (r *Reconciler) createMachineDeployment(ctx context.Context, cluster *clust
 	log := tlog.LoggerFrom(ctx).WithMachineDeployment(md.Object)
 
 	infraCtx, _ := log.WithObject(md.InfrastructureMachineTemplate).Into(ctx)
-	if err := r.reconcileReferencedTemplate(infraCtx, reconcileReferencedTemplateInput{
+	if _, err := r.reconcileReferencedTemplate(infraCtx, reconcileReferencedTemplateInput{
 		cluster: cluster,
 		desired: md.InfrastructureMachineTemplate,
 	}); err != nil {
 		return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: md.Object})
 	}
 
 	bootstrapCtx, _ := log.WithObject(md.BootstrapTemplate).Into(ctx)
-	if err := r.reconcileReferencedTemplate(bootstrapCtx, reconcileReferencedTemplateInput{
+	reconciledMachineDeployment, err := r.reconcileReferencedTemplate(bootstrapCtx, reconcileReferencedTemplateInput{
```

Comment:
I'm a little bit confused by this because the return value of this func is the bootstrapTemplate, not the machineDeployment

Reply:
Going to revert this change.